tests/components/mochad/test_switch.py

```diff
@@ -20,55 +17,38 @@ def pymochad_mock():
         yield
 
 
-class TestMochadSwitchSetup(unittest.TestCase):
-    """Test the mochad switch."""
-
-    PLATFORM = mochad
-    COMPONENT = switch
-    THING = "switch"
-
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.hass.stop)
+@pytest.fixture
+def switch_mock(hass):
+    """Mock switch."""
+    controller_mock = mock.MagicMock()
+    dev_dict = {"address": "a1", "name": "fake_switch"}
+    return mochad.MochadSwitch(hass, controller_mock, dev_dict)
 
-    @mock.patch("homeassistant.components.mochad.switch.MochadSwitch")
-    def test_setup_adds_proper_devices(self, mock_switch):
-        """Test if setup adds devices."""
-        good_config = {
-            "mochad": {},
-            "switch": {
-                "platform": "mochad",
-                "devices": [{"name": "Switch1", "address": "a1"}],
-            },
-        }
-        assert setup_component(self.hass, switch.DOMAIN, good_config)
 
+async def test_setup_adds_proper_devices(hass):
+    """Test if setup adds devices."""
+    good_config = {
+        "mochad": {},
+        "switch": {
+            "platform": "mochad",
+            "devices": [{"name": "Switch1", "address": "a1"}],
+        },
+    }
+    assert await async_setup_component(hass, switch.DOMAIN, good_config)
 
-class TestMochadSwitch(unittest.TestCase):
-    """Test for mochad switch platform."""
 
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        controller_mock = mock.MagicMock()
-        dev_dict = {"address": "a1", "name": "fake_switch"}
-        self.switch = mochad.MochadSwitch(self.hass, controller_mock, dev_dict)
+async def test_name(switch_mock):
+    """Test the name."""
+    assert "fake_switch" == switch_mock.name
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)